### PR TITLE
add material-ui and emotion

### PR DESCRIPTION
**Pull request by interacsean**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,6 +3,10 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@emotion/core": "^10.0.16",
```
Comment (interacsean) [inline package.json]: Lock versions by removing "^"

Comment (interacsean) [inline package.json]: Lock versions by removing "^"

File: package.json [hunk 2]
```diff
@@ -11,15 +15,15 @@
...
     "redux-listeners": "0.4.1",
     "redux-persist": "5.10.0",
     "redux-promise-middleware": "6.1.1",
     "redux-promise-middleware-actions": "2.1.0",
     "redux-thunk": "2.3.0",
-    "typescript": "3.5.1",
-    "redux-devtools-extension": "2.13.7"
+    "typescript": "3.5.1"
   },
   "devDependencies": {
```
Comment (interacsean) [inline package.json:27]: Do @emotion and @material-ui have type definitions (or are the ts definitions included in the core package somehow)?  I.e. try `npm i -D -E @types/@emotion/styled` or something

File: src/themeConfig/palette.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { PaletteOptions } from '@material-ui/core/styles/createPalette';
+import { colorPalette } from './themeConstants';
+
+const {
+  primaryColor,
+  tertiaryColor,
```
Comment (interacsean) [inline src/themeConfig/palette.ts]: ~Would there be a primaryHighlightColor?~

Sorry, I see it is defined, just not imported

File: src/themeConfig/palette.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  tertiaryHighlightColor,
+  quaternaryColor,
+  quaternaryHighlightColor,
+} = colorPalette;
+
+// Just an example, You'll need to configure this for your project.
```
Comment (interacsean) [inline src/themeConfig/palette.ts]: maybe add that this needs to conform to the MaterialUI requirements for a palette?  Dunno, I just remember not realising this was MUI specific

File: src/themeConfig/themeConstants.ts [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+  f1: '3rem',
+  f2: '2.25rem',
+  f3: '1.5rem',
+  f4: '1rem',
+  f5: '.875rem',
+  f6: '.875rem',
+  f7: '.75rem',
+};
+
+export const measure = {
```
Comment (interacsean) [inline src/themeConfig/themeConstants.ts]: Maybe add link to http://tachyons.io/docs/typography/measure/ as comment for lazy devs

File: src/themeConfig/themeConstants.ts [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+  extraExtraExtraLarge: '16rem',
+};
+
+export const widths = {
+  w1: '1rem',
+  w2: '2rem',
+  w3: '4rem',
+  w4: '8rem',
+  w5: '16rem',
+  w10: '10%',
```
Comment (interacsean) [inline src/themeConfig/themeConstants.ts]: I still think we get rid of w10-w100.  

I think having all preferred widths here is a bit of a pipe dream.  w1-w5 I kinda get as general spacing distances.  wAuto I don't see the value of as well.  In use, how different is `width: auto` from `width: ${themeConsts.wAuto}`.  And the value will never change (unlike w1-w5 which could be tweaked without being misleading.)

wThirds will break if what I actually want is `calc(100%/3 - 1rem)` which I often do, to allow for gutters (yes this would be covered by grid hopefully).  

TLDR: I say just keep w1-w5

File: src/themeConfig/themeConstants.ts [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+  mwNone: 'none',
+};
+
+export const heights = {
+  h1: '1rem',
+  h2: '2rem',
+  h3: '4rem',
+  h4: '8rem',
+  h5: '16rem',
+  h25: '25%',
```
Comment (interacsean) [inline src/themeConfig/themeConstants.ts]: Same for height (just keep h1-h5).  Very rarely does specifying height in % actually work anyway.

File: src/themeConfig/themeConstants.ts [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+};
+
+export const borderRadius = {
+  br0: '0',
+  br1: '.125rem',
+  br2: '.25rem',
+  br3: '.5rem',
+  br4: '1rem',
+  br100: '100%',
+  brPill: '9999px',
```
Comment (interacsean) [inline src/themeConfig/themeConstants.ts]: Angels wept

File: src/themeConfig/themeConstants.ts [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+  h5: '16rem',
+  h25: '25%',
+  h50: '50%',
+  h75: '75%',
+  h100: '100%',
+  hAuto: 'auto',
+  hInherit: 'inherit',
+};
+
+export const minHeights = {
```
Comment (interacsean) [inline src/themeConfig/themeConstants.ts]: As per above.

Generally with anything below also:  If it's just an alias for a value that, if the value changes, will be misleading (i.e. imagine changing, then I don't see the point.  Happy for counter-arguments.  Whitelisting 'preferred' values won't go too far, as there'll be exceptions almost immediately and then we just have inconsistent approaches throughout the codebase.

> Note comment at top re converting these to jsInCss for spreading

File: src/themeConfig/themeConstants.ts [hunk 10]
```diff
@@ -0,0 +1,170 @@
+// Basing these constants off Tachyons.
+// An argument could be made for splitting out constants with specific style integrations.
```
Comment (interacsean) [inline src/themeConfig/themeConstants.ts]: I think I prefer this, and it really mirrors more what Tachyon provides – just having the value loses what style rule that value applies to.  I have done it trivially in the past and enjoyed it.

I.e. rather than being a map of values, they are mapped to the style rule.

I.e. instead of `f1: '3rem',`, we have `f1: { fontSize: '3rem', },`
This becomes more useful looking at `minHeight100vh` or whatever.

Tempted to suggest adding stuff for flex, but maybe we hold off and you can play with it in the first project this is used for.

**Review by interacsean — COMMENTED** [review]
Few things to touch up, possibly a change of tack for the style utils
